_posts/en/newsletters/2023-01-18-newsletter.md

```diff
@@ -0,0 +1,216 @@
+    highly trusted spending path, Alice can spend them at any time by
+    satisfying the conditions of that path (e.g. using her cold
+    wallets).
+
+    Beyond proposing the new opcodes, O'Beirne also describes the
+    motivation for vaults and analyzes other vault proposals, including
+    those that are available on Bitcoin now using presigned transactions
+    and those that would depend on other soft fork [covenant][topic
+    covenants] proposals.  Several described advantages of the `OP_VAULT`
+    proposal include:
+
+    - *Smaller witnesses:* flexible covenant proposals, such as those
+      using the proposed [OP_CHECKSIGFROMSTACK][topic
+      op_checksigfromstack], would require that transaction witnesses
+      for unvaulting transactions include copies of a significant amount
+      of the data included elsewhere in the transaction, bloating the
+      size and fee cost of those transactions.  `OP_VAULT` requires much
+      less script and witness data to be included onchain.
+
+    - *Fewer steps for spending:* less flexible covenant proposals and
+      vaults available today based on presigned transactions require
+      withdrawing funds to a predetermined address before they can be sent
+      to an ultimate destination.  Such proposals also generally require
+      each received output be spent in a separate transaction from other
+      received outputs, preventing the use of [payment batching][topic
+      payment batching].  This increases the number of transactions
+      involved, which also bloats the size and cost of spending.
+
+      `OP_VAULT` requires fewer transactions when spending a single
+      output in typical cases and it also supports batching when
+      spending or freezing multiple outputs, potentially saving a large
+      amount of space and allowing vaults to receive many more
+      transactions before their outputs would need to be consolidated
+      for safety.
+
+    In discussion of the idea, Greg Sanders proposed (as [summarized by
+    O'Beirne][obeirne scripthash]) a slightly different construction which
+    "would allow all outputs in vault lifecycles to be [P2TR][topic
+    taproot], for example, which would conceal the operation of the
+    vault---a very nice feature".
+
+    Separately, Anthony Towns [notes][towns op_vault] that the proposal
+    allows the vault user to freeze their funds at any time just by
+    spending the funds to the address for the highly trusted spending
+    path, allowing the user to unfreeze their funds later.  This
+    benefits the vault owner because they don't need to access any
+    specially-secured key material, such as a cold wallet, in order to
+    block a theft attempt.  However, any third party who learns the
+    address can also freeze the user's funds (although they'll have to
+    pay a transaction fee to do so), creating an inconvenience for the
+    user.  Given that many lightweight wallets disclose their addresses
```

Comment:
Is it their XPUBs that they disclose, typically?

Reply:
Depends on the wallet.  For wallets using the Electrum Server Protocol, they disclose a list of individual scriptPubKeys.  For some other wallets, e.g. Samourai, the XPUBs are uploaded.  There are exceptions, such as Neutrino, which uses BIP157/8 compact block filters.